nexus/db-queries/src/db/error.rs

```diff
@@ -93,19 +85,6 @@ fn format_database_error(
     rv
 }
 
-/// Like [`diesel::result::OptionalExtension<T>::optional`]. This turns Ok(v)
-/// into Ok(Some(v)), Err("NotFound") into Ok(None), and leave all other values
-/// unchanged.
-pub fn diesel_result_optional<T>(
-    result: Result<T, ConnectionError>,
-) -> Result<Option<T>, ConnectionError> {
-    match result {
-        Ok(v) => Ok(Some(v)),
-        Err(ConnectionError::Query(DieselError::NotFound)) => Ok(None),
-        Err(e) => Err(e),
-    }
-}
-
```

Comment:
Since we're just using native Diesel errors, this isn't necessary -- we can just use Diesel's built-in `OptionalExtension`.